### PR TITLE
Frost integration 

### PR DESCRIPTION
**Pull request by dmytropiatkivskyi**
Integrates frost signer into btcwallet piggybacking on taproot addresses

### CODE REVIEW

File: btcwallet.go [hunk 1]
```diff
@@ -24,27 +24,38 @@ var (
 	cfg *config
 )
 
-func main() {
+/*func main() {
```
Comment (bullet-tooth) [inline btcwallet.go:27]: please clean code. If we will need to run the wallet as console app then `cmd/btcwallet/main.go` would be a right place for main method

File: btcwallet.go [hunk 2]
```diff
@@ -5,12 +5,12 @@
 package main
```
Comment (bullet-tooth) [inline btcwallet.go:5]: package should be renamed such as it will be used by other apps.

**Review by bullet-tooth — COMMENTED** [review]
what is the purpose of this repo, Showcase how to use btcwallet with frost or make a separate library?

File: signer_with_wallet_example.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/frost"
+	"github.com/btcsuite/btcwallet/waddrmgr"
+	"github.com/btcsuite/btcwallet/wallet"
+	"os"
+	"runtime"
+	"time"
+)
+
+func main() {
```
Comment (bullet-tooth) [inline signer_with_wallet_example.go:16]: nit: by conventions it's better to locate examples in a separate dir, see https://github.com/golang-standards/project-layout/tree/master/examples

**Comment by dmytropiatkivskyi** [comment]
> what is the purpose of this repo, Showcase how to use btcwallet with frost or make a separate library?

the main purpose is to integrate frost into the wallet. Binaries are to be generated and the modified wallet is to be used as a lib.

File: btcwallet.go (hunk 2 above)
Comment (dmytropiatkivskyi) [reply btcwallet.go:5]: the idea was to do as little changes to the wallet's code as possible. the main package won't be imported anyway, but other packages
does this solve your concern?

File: signer_with_wallet_example.go (hunk 3 above)
Comment (dmytropiatkivskyi) [reply signer_with_wallet_example.go:16]: the example uses the code from the main package, thus has to remain there (if we want to change the original code as little as possible)

**Review by bullet-tooth — APPROVED** [review]

**Pull request merged**